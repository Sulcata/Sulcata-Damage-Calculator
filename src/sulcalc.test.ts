import { NoMoveError, NoPokemonError } from "./errors";
import sulcalc from "./sulcalc";
import {
  Generation,
  Nature,
  Status,
  Terrain,
  Weather,
  generations,
  maxGen
} from "./utilities";

test("sanity check", () => {
  for (const gen of generations) {
    const { summary, damage } = sulcalc(
      { name: "Snorlax", gen },
      { name: "Snorlax", gen },
      { name: "Body Slam", gen },
      { gen }
    );
    expect(summary).toMatchSnapshot();
    expect(damage).toMatchSnapshot();
  }
});

test("bail out on insufficient damage", () => {
  const gen = maxGen;
  const { summary } = sulcalc(
    { name: "Munchlax", gen },
    { name: "Aggron", gen },
    { name: "Pound", gen },
    { gen }
  );
  expect(summary).toMatchSnapshot();
});

test("guaranteed KO", () => {
  const gen = Generation.GSC;
  const { summary } = sulcalc(
    { name: "Snorlax", gen },
    { name: "Raikou", gen },
    { name: "Self-Destruct", gen },
    { gen }
  );
  expect(summary).toMatchSnapshot();
});

test("almost guaranteed KO", () => {
  const gen = Generation.GSC;
  const { summary } = sulcalc(
    {
      name: "Snorlax",
      evs: [252, 200, 252, 252, 252, 252],
      gen
    },
    { name: "Snorlax", gen },
    { name: "Double-Edge", gen },
    { gen }
  );
  expect(summary).toMatchSnapshot();
});

test("negligible chance to KO", () => {
  const gen = Generation.GSC;
  const { summary } = sulcalc(
    {
      name: "Tauros",
      evs: [252, 176, 252, 252, 252, 252],
      gen
    },
    {
      name: "Snorlax",
      evs: [252, 252, 252, 252, 252, 252],
      gen
    },
    { name: "Body Slam", gen },
    { gen }
  );
  expect(summary).toMatchSnapshot();
});

test("Psyshock", () => {
  const gen = maxGen;
  const { summary } = sulcalc(
    {
      name: "Latios",
      item: "Life Orb",
      nature: Nature.TIMID,
      evs: [4, 0, 0, 252, 0, 252],
      gen
    },
    {
      name: "Chansey",
      item: "Eviolite",
      nature: Nature.BOLD,
      evs: [252, 0, 252, 0, 4, 0],
      gen
    },
    { name: "Psyshock", gen },
    { gen }
  );
  expect(summary).toMatchSnapshot();
});

test("Ability", () => {
  const gen = maxGen;
  const { summary } = sulcalc(
    {
      name: "Heracross",
      ability: "Guts",
      nature: Nature.ADAMANT,
      evs: [0, 252, 4, 0, 0, 252],
      boosts: [0, 2, 0, 0, 0, 0, 0, 0],
      status: Status.BURNED
    },
    {
      name: "Skarmory",
      ability: "Sturdy",
      nature: Nature.IMPISH,
      evs: [248, 0, 252, 0, 8, 0]
    },
    {
      name: "Close Combat",
      critical: true,
      gen
    },
    { gen }
  );
  expect(summary).toMatchSnapshot();
});

test("Explosion", () => {
  const gen = maxGen;
  const { roundedChances } = sulcalc(
    { name: "Snorlax", gen },
    { name: "Snorlax", gen },
    { name: "Explosion", gen },
    { gen }
  );
  expect(roundedChances).toHaveLength(1);
  const { summary } = sulcalc(
    { name: "Snorlax", gen },
    { name: "Skarmory", gen },
    { name: "Explosion", gen },
    { gen }
  );
  expect(summary).toMatchSnapshot();
});

test("throws NoPokemonError when either Pokemon is invalid", () => {
  expect(() => {
    sulcalc({ id: "0:0" }, { id: "0:0" }, { id: "1" }, {});
  }).toThrow(NoPokemonError);

  expect(() => {
    sulcalc({ id: "233" }, { id: "233" }, { id: "1" }, {});
  }).toThrow(NoPokemonError);
});

test("throws NoMoveError when either Move is invalid", () => {
  expect(() => {
    sulcalc({ name: "Snorlax" }, { name: "Snorlax" }, { id: "0" }, {});
  }).toThrow(NoMoveError);
});

test("Variable BP fix", () => {
  const gen = Generation.GSC;
  const { summary } = sulcalc(
    {
      name: "Gyarados",
      item: "Polkadot Bow",
      currentHp: 15,
      gen
    },
    { name: "Articuno", gen },
    { name: "Flail", gen },
    { gen }
  );
  expect(summary).toMatchSnapshot();
});

test("Present GSC", () => {
  const gen = Generation.GSC;
  for (const defenderName of ["Raikou", "Vaporeon", "Zapdos"]) {
    for (let present = -1; present <= 3; present++) {
      expect(
        sulcalc(
          { name: "Blissey", gen },
          { name: defenderName, gen },
          { name: "Present", present, gen },
          { gen }
        ).summary
      ).toMatchSnapshot();
    }
  }
});

test("Terrains boost types", () => {
  for (const gen of generations.slice(Generation.B2W2)) {
    expect(
      sulcalc(
        {
          name: "Tapu Lele",
          nature: Nature.TIMID,
          evs: [0, 0, 0, 252, 4, 252],
          gen
        },
        {
          name: "Landorus-Therian",
          nature: Nature.ADAMANT,
          evs: [0, 252, 4, 0, 0, 252],
          gen
        },
        { name: "Psychic", gen },
        { terrain: Terrain.PSYCHIC_TERRAIN, gen }
      ).summary
    ).toMatchSnapshot();

    expect(
      sulcalc(
        {
          name: "Tapu Koko",
          nature: Nature.NAIVE,
          evs: [0, 252, 0, 4, 0, 252],
          gen
        },
        {
          name: "Diancie",
          nature: Nature.NAIVE,
          evs: [0, 4, 0, 252, 0, 252],
          gen
        },
        { name: "Wild Charge", gen },
        { terrain: Terrain.ELECTRIC_TERRAIN, gen }
      ).summary
    ).toMatchSnapshot();

    expect(
      sulcalc(
        {
          name: "Tapu Bulu",
          item: "Choice Band",
          nature: Nature.ADAMANT,
          evs: [0, 252, 4, 0, 0, 252],
          boosts: [0, -1, 0, 0, 0, 0, 0, 0],
          gen
        },
        {
          name: "Landorus-Therian",
          nature: Nature.IMPISH,
          evs: [252, 0, 216, 0, 24, 16],
          gen
        },
        { name: "Wood Hammer", gen },
        { terrain: Terrain.GRASSY_TERRAIN, gen }
      ).summary
    ).toMatchSnapshot();
  }
});

test("Greninja-Ash Water Shuriken", () => {
  const gen = maxGen;
  expect(
    sulcalc(
      {
        name: "Greninja-Ash",
        item: "Choice Specs",
        nature: Nature.TIMID,
        evs: [0, 0, 4, 252, 0, 252],
        gen
      },
      {
        name: "Tapu Koko",
        nature: Nature.NAIVE,
        evs: [0, 252, 0, 4, 0, 252],
        gen
      },
      { name: "Water Shuriken", gen },
      { gen }
    ).summary
  ).toMatchSnapshot();
});

test("Skill Link", () => {
  const gen = maxGen;
  expect(
    sulcalc(
      {
        name: "Cloyster",
        nature: Nature.NAUGHTY,
        ability: "Skill Link",
        evs: [0, 252, 4, 0, 0, 252],
        gen
      },
      {
        name: "Rotom-Wash",
        nature: Nature.BOLD,
        evs: [248, 0, 200, 0, 0, 60],
        gen
      },
      { name: "Icicle Spear", gen },
      { gen }
    ).summary
  ).toMatchSnapshot();
});

test("Thousand Arrows", () => {
  const gen = maxGen;
  expect(
    sulcalc(
      {
        name: "Zygarde",
        item: "Choice Band",
        nature: Nature.ADAMANT,
        evs: [4, 252, 0, 0, 0, 252],
        gen
      },
      {
        name: "Rotom-Wash",
        nature: Nature.BOLD,
        ability: "Levitate",
        evs: [248, 0, 200, 0, 0, 60],
        gen
      },
      { name: "Thousand Arrows", gen },
      { gen }
    ).summary
  ).toMatchSnapshot();
});

test("Levitate immunity to Ground moves", () => {
  const gen = maxGen;
  expect(
    sulcalc(
      {
        name: "Zygarde",
        item: "Choice Band",
        nature: Nature.ADAMANT,
        evs: [4, 252, 0, 0, 0, 252],
        gen
      },
      {
        name: "Rotom-Wash",
        nature: Nature.BOLD,
        ability: "Levitate",
        evs: [248, 0, 200, 0, 0, 60],
        gen
      },
      { name: "Earthquake", gen },
      { gen }
    ).summary
  ).toMatchSnapshot();
});

test("Grassy Terrain recovery", () => {
  const gen = maxGen;
  expect(
    sulcalc(
      {
        name: "Tapu Bulu",
        nature: Nature.JOLLY,
        evs: [0, 252, 4, 0, 0, 252],
        gen
      },
      {
        name: "Chansey",
        item: "Eviolite",
        nature: Nature.BOLD,
        evs: [248, 0, 252, 0, 8, 0],
        gen
      },
      { name: "Horn Leech", gen },
      { terrain: Terrain.GRASSY_TERRAIN, gen }
    ).summary
  ).toMatchSnapshot();
});

test("Defender health text shows for not completely healthy Pokemon", () => {
  const gen = Generation.RBY;
  expect(
    sulcalc(
      { name: "Tauros", gen },
      { name: "Snorlax", currentHp: 40, gen },
      { name: "Hyper Beam", gen },
      { gen }
    ).summary
  ).toContain("at 7%");
});

test("Arceus Judgment with plate", () => {
  const gen = maxGen;
  expect(
    sulcalc(
      {
        name: "Arceus-Dark",
        item: "Dread Plate",
        evs: [0, 0, 0, 8, 0, 0],
        gen
      },
      {
        name: "Abomasnow",
        evs: [252, 0, 0, 0, 0, 0],
        gen
      },
      { name: "Judgment", gen },
      { gen }
    ).summary
  ).toMatchSnapshot();
});

test("Weather is reported during Pokemon description", () => {
  const gen = Generation.GSC;
  expect(
    sulcalc(
      { name: "Moltres", gen },
      { name: "Snorlax", gen },
      { name: "Fire Blast", gen },
      { weather: Weather.SUN, gen }
    ).summary
  ).toMatch(/^.+? vs\. .+? in Sun: /);
});

test("Berry damage", () => {
  const gen = Generation.GSC;

  const { roundedChances: chancesWithoutBerry } = sulcalc(
    { name: "Pichu", gen },
    { name: "Pichu", gen },
    { name: "Double-Edge", gen },
    { gen }
  );
  expect(chancesWithoutBerry[0]).toBe(0);
  expect(chancesWithoutBerry[1]).toBeCloseTo(0.993, 3);
  expect(chancesWithoutBerry[2]).toBe(1);

  const { roundedChances: chancesWithBerry } = sulcalc(
    { name: "Pichu", gen },
    { name: "Pichu", item: "Berry", gen },
    { name: "Double-Edge", gen },
    { gen }
  );
  expect(chancesWithBerry[0]).toBe(0);
  expect(chancesWithBerry[1]).toBeCloseTo(0.871, 3);
  expect(chancesWithBerry[2]).toBe(1);
});

describe("toxic damage", () => {
  const gen = Generation.B2W2;
  const { roundedChances } = sulcalc(
    { name: "Pichu", gen },
    { name: "Pichu", status: Status.BADLY_POISONED, gen },
    { name: "Attract", gen },
    { gen }
  );
  expect(roundedChances).toEqual([0, 0, 0, 0, 0, 1]);
});

describe("stealth rock damage", () => {
  const gen = Generation.B2W2;

  const { roundedChances: withoutRocks } = sulcalc(
    { name: "Starmie", gen },
    { name: "Volcarona", gen },
    { name: "Bubble Beam", gen },
    { gen }
  );
  expect(withoutRocks[0]).toBe(0);
  expect(withoutRocks[1]).toBeCloseTo(0.094, 3);
  expect(withoutRocks[2]).toBe(1);

  const { roundedChances: withRocks } = sulcalc(
    { name: "Starmie", gen },
    { name: "Volcarona", stealthRock: true, gen },
    { name: "Bubble Beam", gen },
    { gen }
  );
  expect(withRocks).toEqual([0.25, 1]);
});

test("Displays boosts", () => {
  const gen = Generation.RBY;

  expect(
    sulcalc(
      { name: "Alakazam", boosts: [0, 0, 0, 1, 0, 0, 0, 0], gen },
      { name: "Jynx", boosts: [0, 0, 0, -1, 0, 0, 0, 0], gen },
      { name: "Psychic", gen },
      { gen }
    ).summary
  ).toMatch(/^\+1 Alakazam .+ -1 Jynx/);

  expect(
    sulcalc(
      { name: "Tauros", boosts: [0, 1, 0, 0, 0, 0, 0, 0], gen },
      { name: "Jynx", boosts: [0, 0, 4, 0, 0, 0, 0, 0], gen },
      { name: "Body Slam", gen },
      { gen }
    ).summary
  ).toMatch(/^\+1 Tauros .+ \+4 Jynx/);
});

test("Light Screen", () => {
  const gen = Generation.RBY;
  expect(
    sulcalc(
      { name: "Alakazam", gen },
      { name: "Jynx", lightScreen: true, gen },
      { name: "Psychic", gen },
      { gen }
    ).summary
  ).toMatch(/^Alakazam .+ Jynx behind Light Screen/);
});

test("Reflect", () => {
  const gen = Generation.RBY;
  expect(
    sulcalc(
      { name: "Tauros", gen },
      { name: "Jynx", reflect: true, gen },
      { name: "Body Slam", gen },
      { gen }
    ).summary
  ).toMatch(/^Tauros .+ Jynx behind Reflect/);
});
