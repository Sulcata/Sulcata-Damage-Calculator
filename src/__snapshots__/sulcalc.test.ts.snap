// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ability 1`] = `"+2 252+ Atk Guts Heracross (burned) critical hit Close Combat vs. 248 HP / 252+ Def Sturdy Skarmory: 332 (99.7%) -- guaranteed 2HKO"`;

exports[`Arceus Judgment with plate 1`] = `"8 SpA Dread Plate Arceus-Dark Judgment vs. 252 HP / 0 SpD Abomasnow: 175 - 207 (45.6 - 53.9%) -- 43.8% chance to 2HKO"`;

exports[`Explosion 1`] = `"0 Atk Snorlax Explosion vs. 0 HP / 0 Def Skarmory: 109 - 129 (40.2 - 47.6%) -- this might take a while..."`;

exports[`Grassy Terrain recovery 1`] = `"252 Atk Tapu Bulu Horn Leech vs. 248 HP / 252+ Def Eviolite Chansey: 246 - 289 (35 - 41.1%) -- 68.3% chance to 3HKO after Grassy Terrain"`;

exports[`Greninja-Ash Water Shuriken 1`] = `"252 SpA Choice Specs Greninja-Ash Water Shuriken [3 hits] vs. 0 HP / 0- SpD Tapu Koko: 237 - 282 (84.3 - 100.4%) -- 0.1% chance to OHKO"`;

exports[`Levitate immunity to Ground moves 1`] = `"252+ Atk Choice Band Zygarde Earthquake vs. 248 HP / 200+ Def Levitate Rotom-Wash: 0 (0%) -- this might take a while..."`;

exports[`Present GSC 1`] = `"Blissey Present vs. Raikou: -95 - 399 (-24.8 - 104.2%) -- 2.9% chance to OHKO, 30.3% chance to 2HKO, 61.9% chance to 3HKO, 81.9% chance to 4HKO, 91.3% chance to 5HKO, 95.6% chance to 6HKO, 97.7% chance to 7HKO, 98.8% chance to 8HKO, 99.4% chance to 9HKO"`;

exports[`Present GSC 2`] = `"Blissey Present vs. Raikou: -95 (-24.8%) -- this might take a while..."`;

exports[`Present GSC 3`] = `"Blissey Present vs. Raikou: 114 - 135 (29.8 - 35.2%) -- 19.8% chance to 3HKO"`;

exports[`Present GSC 4`] = `"Blissey Present vs. Raikou: 227 - 267 (59.3 - 69.7%) -- guaranteed 2HKO"`;

exports[`Present GSC 5`] = `"Blissey Present vs. Raikou: 339 - 399 (88.5 - 104.2%) -- 28.2% chance to OHKO"`;

exports[`Present GSC 6`] = `"Blissey Present vs. Vaporeon: -115 - 363 (-24.8 - 78.4%) -- 10.9% chance to 2HKO, 38.1% chance to 3HKO, 60.8% chance to 4HKO, 75.9% chance to 5HKO, 85.5% chance to 6HKO, 91.3% chance to 7HKO, 94.9% chance to 8HKO, 97% chance to 9HKO"`;

exports[`Present GSC 7`] = `"Blissey Present vs. Vaporeon: -115 (-24.8%) -- this might take a while..."`;

exports[`Present GSC 8`] = `"Blissey Present vs. Vaporeon: 104 - 123 (22.5 - 26.6%) -- 21.3% chance to 4HKO"`;

exports[`Present GSC 9`] = `"Blissey Present vs. Vaporeon: 206 - 243 (44.5 - 52.5%) -- 20.1% chance to 2HKO"`;

exports[`Present GSC 10`] = `"Blissey Present vs. Vaporeon: 308 - 363 (66.5 - 78.4%) -- guaranteed 2HKO"`;

exports[`Present GSC 11`] = `"Blissey Present vs. Zapdos: -95 - 75 (-24.8 - 19.6%) -- negligible chance to 6HKO, 0.1% chance to 7HKO, 0.9% chance to 8HKO, 3.1% chance to 9HKO"`;

exports[`Present GSC 12`] = `"Blissey Present vs. Zapdos: -95 (-24.8%) -- this might take a while..."`;

exports[`Present GSC 13`] = `"Blissey Present vs. Zapdos: 22 - 27 (5.7 - 7%) -- this might take a while..."`;

exports[`Present GSC 14`] = `"Blissey Present vs. Zapdos: 43 - 51 (11.2 - 13.3%) -- 10.5% chance to 8HKO"`;

exports[`Present GSC 15`] = `"Blissey Present vs. Zapdos: 63 - 75 (16.4 - 19.6%) -- almost guaranteed 6HKO"`;

exports[`Psyshock 1`] = `"252 SpA Life Orb Latios Psyshock vs. 252 HP / 252+ Def Eviolite Chansey: 226 - 266 (32.1 - 37.8%) -- 93.8% chance to 3HKO"`;

exports[`Skill Link 1`] = `"252+ Atk Skill Link Cloyster Icicle Spear [5 hits] vs. 248 HP / 200+ Def Rotom-Wash: 65 - 80 (21.5 - 26.4%) -- negligible chance to 4HKO"`;

exports[`Terrains boost types 1`] = `"252 SpA Tapu Lele Psychic vs. 0 HP / 0 SpD Landorus-Therian: 267 - 315 (83.7 - 98.7%) -- guaranteed 2HKO"`;

exports[`Terrains boost types 2`] = `"252 Atk Tapu Koko Wild Charge vs. 0 HP / 0 Def Diancie: 144 - 171 (59.8 - 71%) -- guaranteed 2HKO"`;

exports[`Terrains boost types 3`] = `"-1 252+ Atk Choice Band Tapu Bulu Wood Hammer vs. 252 HP / 216+ Def Landorus-Therian: 256 - 303 (67 - 79.3%) -- guaranteed 2HKO"`;

exports[`Terrains boost types 4`] = `"252 SpA Tapu Lele Psychic vs. 0 HP / 0 SpD Landorus-Therian: 267 - 315 (83.7 - 98.7%) -- guaranteed 2HKO"`;

exports[`Terrains boost types 5`] = `"252 Atk Tapu Koko Wild Charge vs. 0 HP / 0 Def Diancie: 144 - 171 (59.8 - 71%) -- guaranteed 2HKO"`;

exports[`Terrains boost types 6`] = `"-1 252+ Atk Choice Band Tapu Bulu Wood Hammer vs. 252 HP / 216+ Def Landorus-Therian: 256 - 303 (67 - 79.3%) -- guaranteed 2HKO"`;

exports[`Thousand Arrows 1`] = `"252+ Atk Choice Band Zygarde Thousand Arrows vs. 248 HP / 200+ Def Levitate Rotom-Wash: 288 - 342 (95 - 112.9%) -- 68.8% chance to OHKO"`;

exports[`Variable BP fix 1`] = `"Polkadot Bow Gyarados Flail vs. Articuno: 218 (56.9%) -- guaranteed 2HKO"`;

exports[`almost guaranteed KO 1`] = `"200 Atk Snorlax Double-Edge vs. Snorlax: 173 - 204 (33.1 - 39%) -- almost guaranteed 3HKO"`;

exports[`bail out on insufficient damage 1`] = `"0 Atk Munchlax Pound vs. 0 HP / 0 Def Aggron: 6 - 7 (2.1 - 2.5%) -- this might take a while..."`;

exports[`guaranteed KO 1`] = `"Snorlax Self-Destruct vs. Raikou: 548 - 645 (143.1 - 168.4%) -- guaranteed OHKO"`;

exports[`negligible chance to KO 1`] = `"176 Atk Tauros Body Slam vs. Snorlax: 112 - 132 (21.4 - 25.2%) -- negligible chance to 4HKO"`;

exports[`sanity check 1`] = `"Snorlax Body Slam vs. Snorlax: 127 - 150 (24.3 - 28.7%) -- 99.2% chance to 4HKO"`;

exports[`sanity check 2`] = `
Multiset {
  "data": Map {
    127 => "1",
    128 => "2",
    129 => "1",
    130 => "2",
    131 => "2",
    132 => "2",
    133 => "1",
    134 => "2",
    135 => "2",
    136 => "1",
    137 => "2",
    138 => "2",
    139 => "1",
    140 => "2",
    141 => "2",
    142 => "2",
    143 => "1",
    144 => "2",
    145 => "2",
    146 => "1",
    147 => "2",
    148 => "2",
    149 => "1",
    150 => "1",
  },
}
`;

exports[`sanity check 3`] = `"Snorlax Body Slam vs. Snorlax: 127 - 150 (24.3 - 28.7%) -- 99.2% chance to 4HKO"`;

exports[`sanity check 4`] = `
Multiset {
  "data": Map {
    127 => "1",
    128 => "2",
    129 => "1",
    130 => "2",
    131 => "2",
    132 => "2",
    133 => "1",
    134 => "2",
    135 => "2",
    136 => "1",
    137 => "2",
    138 => "2",
    139 => "1",
    140 => "2",
    141 => "2",
    142 => "2",
    143 => "1",
    144 => "2",
    145 => "2",
    146 => "1",
    147 => "2",
    148 => "2",
    149 => "1",
    150 => "1",
  },
}
`;

exports[`sanity check 5`] = `"0 Atk Snorlax Body Slam vs. 0 HP / 0 Def Snorlax: 142 - 168 (30.8 - 36.4%) -- 61.1% chance to 3HKO"`;

exports[`sanity check 6`] = `
Multiset {
  "data": Map {
    142 => "1",
    144 => "1",
    146 => "1",
    147 => "1",
    149 => "1",
    151 => "1",
    152 => "1",
    154 => "1",
    156 => "1",
    157 => "1",
    159 => "1",
    161 => "1",
    162 => "1",
    164 => "1",
    166 => "1",
    168 => "1",
  },
}
`;

exports[`sanity check 7`] = `"0 Atk Snorlax Body Slam vs. 0 HP / 0 Def Snorlax: 142 - 168 (30.8 - 36.4%) -- 57.8% chance to 3HKO"`;

exports[`sanity check 8`] = `
Multiset {
  "data": Map {
    142 => "1",
    144 => "1",
    145 => "1",
    147 => "1",
    148 => "1",
    150 => "1",
    151 => "1",
    154 => "1",
    156 => "1",
    157 => "1",
    159 => "1",
    160 => "1",
    162 => "1",
    163 => "1",
    165 => "1",
    168 => "1",
  },
}
`;

exports[`sanity check 9`] = `"0 Atk Snorlax Body Slam vs. 0 HP / 0 Def Snorlax: 142 - 168 (30.8 - 36.4%) -- 57.8% chance to 3HKO"`;

exports[`sanity check 10`] = `
Multiset {
  "data": Map {
    142 => "1",
    144 => "1",
    145 => "1",
    147 => "1",
    148 => "1",
    150 => "1",
    151 => "1",
    154 => "1",
    156 => "1",
    157 => "1",
    159 => "1",
    160 => "1",
    162 => "1",
    163 => "1",
    165 => "1",
    168 => "1",
  },
}
`;

exports[`sanity check 11`] = `"0 Atk Snorlax Body Slam vs. 0 HP / 0 Def Snorlax: 142 - 168 (30.8 - 36.4%) -- 57.8% chance to 3HKO"`;

exports[`sanity check 12`] = `
Multiset {
  "data": Map {
    142 => "1",
    144 => "1",
    145 => "1",
    147 => "1",
    148 => "1",
    150 => "1",
    151 => "1",
    154 => "1",
    156 => "1",
    157 => "1",
    159 => "1",
    160 => "1",
    162 => "1",
    163 => "1",
    165 => "1",
    168 => "1",
  },
}
`;

exports[`sanity check 13`] = `"0 Atk Snorlax Body Slam vs. 0 HP / 0 Def Snorlax: 142 - 168 (30.8 - 36.4%) -- 57.8% chance to 3HKO"`;

exports[`sanity check 14`] = `
Multiset {
  "data": Map {
    142 => "1",
    144 => "1",
    145 => "1",
    147 => "1",
    148 => "1",
    150 => "1",
    151 => "1",
    154 => "1",
    156 => "1",
    157 => "1",
    159 => "1",
    160 => "1",
    162 => "1",
    163 => "1",
    165 => "1",
    168 => "1",
  },
}
`;
